api/client/cli.go

```diff
@@ -91,19 +91,25 @@ func (cli *DockerCli) Subcmd(name, signature, description string, exitOnError bo
 		errorHandling = flag.ContinueOnError
 	}
 	flags := flag.NewFlagSet(name, errorHandling)
+	options := ""
+	if signature != "" {
+		signature = " " + signature
+	}
+	if flags.FlagCountUndeprecated() > 0 {
+		options = " [OPTIONS]"
+	} else {
+		options = ""
+	}
 	flags.Usage = func() {
-		options := ""
-		if signature != "" {
-			signature = " " + signature
-		}
-		if flags.FlagCountUndeprecated() > 0 {
-			options = " [OPTIONS]"
-		}
 		fmt.Fprintf(cli.out, "\nUsage: docker %s%s%s\n\n%s\n\n", name, options, signature, description)
```

Comment:
To ensure consistency, should these 2 lines be replaced with a call to `flags.ShortUsage()` ?